sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/finetuning/finetuning_vertical.py

```diff
@@ -0,0 +1,193 @@
+from typing import Any, Optional, cast
+from azure.ai.ml import Input
+from azure.ai.ml.exceptions import ErrorCategory, ErrorTarget, ValidationException
+from azure.ai.ml._restclient.v2024_01_01_preview.models import (
+    ModelProvider as RestModelProvider,
+    FineTuningVertical as RestFineTuningVertical,
+    UriFileJobInput,
+    MLFlowModelJobInput,
+)
+from azure.ai.ml.constants._common import AssetTypes
+from azure.ai.ml._utils.utils import camel_to_snake
+from azure.ai.ml.entities._job.finetuning.finetuning_job import FineTuningJob
+
+
+class FineTuningVertical(FineTuningJob):
+    def __init__(
+        self,
+        *,
+        task: str,
+        model: Input,
+        model_provider: str,
+        training_data: Input,
+        validation_data: Optional[Input] = None,
+        **kwargs: Any,
+    ) -> None:
+        self._task = task
+        self._model = model
+        self._model_provider = model_provider
+        self._training_data = training_data
+        self._validation_data = validation_data
+        super().__init__(**kwargs)
+
+    @property
+    def task(self) -> str:
+        """Get finetuning task.
+
+        :return: The type of task to run. Possible values include: "ChatCompletion"
+                 "TextCompletion", "TextClassification", "QuestionAnswering","TextSummarization",
+                 "TokenClassification", "TextTranslation", "ImageClassification", "ImageInstanceSegmentation",
+                 "ImageObjectDetection","VideoMultiObjectTracking".
+
+        :rtype: str
+        """
+        return self._task
+
+    @task.setter
+    def task(self, task: str) -> None:
+        """Set finetuning task.
+
+        :param task_type: The type of task to run. Possible values include: "ChatCompletion"
+                 "TextCompletion", "TextClassification", "QuestionAnswering","TextSummarization",
+                 "TokenClassification", "TextTranslation", "ImageClassification", "ImageInstanceSegmentation",
+                 "ImageObjectDetection","VideoMultiObjectTracking",.
+
+        :type task: str
+        """
+        self._task = task
+
+    @property
+    def model(self) -> Optional[Input]:
+        """The model to be fine-tuned."""
+        return self._model
+
+    @model.setter
+    def model(self, value: Input) -> None:
+        """Set the model to be fine-tuned.
+
+        :param value: Input object representing the mlflow model to be fine-tuned.
+        :type value: typing.Union[typing.Dict, TabularLimitSettings]
+        :raises ValidationException: Expected a mlflow model input.
+        """
+        if isinstance(value, Input) and cast(Input, value).type == "mlflow_model":
+            self._model = value
+        else:
+            if not isinstance(value, dict):
```

Comment:
curious why another if condition inside else, what is expected when   isinstance(value, dict) is true?

Reply:
I think it was unintentional, corrected it, thanks for catching it.